packages/core/package-manager/src/JSONParseStream.js

```diff
@@ -17,7 +17,13 @@ export default class JSONParseStream extends Transform {
     callback: (err: ?Error, parsed: ?JSONObject) => mixed,
   ) {
     try {
-      callback(null, JSON.parse(chunk.toString()));
+      let parsed;
+      try {
+        parsed = JSON.parse(chunk.toString());
+      } catch (e) {
+        return;
```

Comment:
Maybe a comment here that we're intentionally not calling the callback with these errors because some streaming JSON implementations leak non-json?

Reply:
Exactly (this specific case in Yarn 2 will be fixed by https://github.com/yarnpkg/berry/pull/2883), I've added the comment.

Good idea to log it